api/src/routers/meals.js

```diff
@@ -0,0 +1,204 @@
+    price,
+    created_date,
+  } = req.body;
+  if (
+    !title ||
+    !description ||
+    !location ||
+    !when ||
+    !max_reservations ||
+    !price ||
+    !created_date
+  ) {
+    return res.status(400).json({
+      error: "All fields are required for posting a meal :(",
+      requiredFields: {
+        title: "string",
+        description: "string",
+        location: "string",
+        when: "datetime (YYYY-MM-DD HH:MM:SS)",
+        max_reservations: "integer",
+        price: "decimal",
+        created_date: "date (YYYY-MM-DD)",
+      },
+    });
+  }
+
+  // Format when (datetime) and created_date (date)
+  const parsedWhenDate = new Date(when);
+  if (isNaN(parsedWhenDate.getTime())) {
+    return res
+      .status(400)
+      .json({ error: "Invalid date format. Use YYYY-MM-DD HH:MM:SS format." });
+  }
+  const formattedWhenDate = parsedWhenDate
+    .toISOString()
+    .slice(0, 19)
+    .replace("T", " ");
+  const parsedCreatedDate = new Date(created_date);
+  if (isNaN(parsedCreatedDate.getTime())) {
+    return res
+      .status(400)
+      .json({ error: "Invalid 'created_date' format. Use YYYY-MM-DD format." });
+  }
+  const formattedCreatedDate = parsedCreatedDate.toISOString().slice(0, 10);
+
+  try {
+    const [newMeal] = await knex("Meal").insert({
+      title,
+      description,
+      location,
+      when: formattedWhenDate,
+      max_reservations,
+      price,
+      created_date: formattedCreatedDate,
+    });
+
+    // Respond with the new added meal
+    res.status(201).json({
+      message: `${title} was added :)`,
+      id: newMeal,
+      title,
+      description,
+      location,
+      when: formattedWhenDate,
+      max_reservations,
+      price,
+      created_date: formattedCreatedDate,
+    });
+  } catch (error) {
+    const errMessage = error.message;
+    res.status(500).json({ error: errMessage });
+  }
+});
+// ----------- /api/meals | POST | Adds a new meal to the database -----------
+
+// ----------- /api/meals/:id | GET | Returns the meal by id -----------
+Meals.get("/:id", async (req, res) => {
+  const id = parseInt(req.params.id);
+  if (isNaN(id)) {
+    return res.status(400).send("Invalid Id");
+  }
+  try {
+    const meal = await knex("Meal").where("id", id);
+
+    if (meal) {
+      return res.json(meal);
+    } else {
+      return res.status(404).send("Meal Not Found");
+    }
+  } catch (error) {
+    const errMessage = error.message;
+    res.status(500).json({ error: errMessage });
+  }
+});
+// ----------- /api/meals/:id | GET | Returns the meal by id -----------
+
+// ----------- /api/meals/:id | PUT | Updates the meal by id -----------
+Meals.put("/:id", async (req, res) => {
+  const id = parseInt(req.params.id);
+  const {
+    title,
+    description,
+    location,
+    when,
+    max_reservations,
+    price,
+    created_date,
+  } = req.body;
+
+  const updateFields = {};
+
+  if (title !== undefined) updateFields.title = title;
```

Comment:
again nitpicking, but you seem to handle everything really well in terms of material so I can maybe slide in some more details and know-how :) feel free to ignore it.
It's again in terms of convetion or good practice to name your constants as nouns and functions as verbs. In this case by `updateFields` one might expect a function that does the updating. In your case, perhaps `fieldsToUpdate` might be a better naming. Does this make sense?

Keeping to the conventions reduces mental overhead when maintaining your code! :)